sdks/python/apache_beam/transforms/external.py

```diff
@@ -144,6 +147,136 @@ def _get_named_tuple_instance(self):
+      named_fields.append(
+          (key, convert_to_typing_type(instance_to_type(value))))
+
+    schema_proto = named_fields_to_schema(named_fields)
+    row = named_tuple_from_schema(schema_proto)(**kwargs)
+    schema = named_tuple_to_schema(type(row))
+
+    payload = RowCoder(schema).encode(row)
+    return (schema_proto, payload)
+
+  def build(self):
+    constructor_params = self._constructor_params or {}
+    constructor_schema, constructor_payload = (
+        self._get_schema_proto_and_payload(**constructor_params))
+    payload = JavaClassLookupPayload(
+        class_name=self._class_name,
+        constructor_schema=constructor_schema,
+        constructor_payload=constructor_payload)
+    if self._constructor_method:
+      payload.constructor_method = self._constructor_method
+
+    for builder_method_name, params in self._builder_methods_and_params.items():
+      builder_method_schema, builder_method_payload = (
+          self._get_schema_proto_and_payload(**params))
+      builder_method = BuilderMethod(
+          name=builder_method_name,
+          schema=builder_method_schema,
+          payload=builder_method_payload)
+      builder_method.name = builder_method_name
+      payload.builder_methods.append(builder_method)
+    return payload
+
+  def add_constructor(self, **kwargs):
+    """
+    Specifies the Java constructor to use.
+
+    :param kwargs: parameter names and values of the constructor.
+    """
+    if self._constructor_method or self._constructor_params:
+      raise ValueError(
+          'Constructor or constructor method can only be specified once')
+
+    self._constructor_params = kwargs
+
+  def add_constructor_method(self, method_name, **kwargs):
+    """
+    Specifies the Java constructor method to use.
+
+    :param method_name: name of the constructor method.
+    :param kwargs: parameter names and values of the constructor method.
+    """
+    if self._constructor_method or self._constructor_params:
+      raise ValueError(
+          'Constructor or constructor method can only be specified once')
+
+    self._constructor_method = method_name
+    self._constructor_params = kwargs
+
+  def add_builder_method(self, method_name, **kwargs):
+    """
+    Specifies a Java builder method to be invoked after instantiating the Java
+    transform class. Specified builder method will be applied in order.
+
+    :param method_name: name of the builder method.
+    :param kwargs:  parameter names and values of the builder method.
+    """
+    self._builder_methods_and_params[method_name] = kwargs
+
+
+class JavaExternalTransform(ptransform.PTransform):
+  """A proxy for Java-implemented external transforms.
+
+  One builds these transforms just as one would in Java.
+  """
+  def __init__(self, class_name, expansion_service=None):
+    self._payload_builder = JavaClassLookupPayloadBuilder(class_name)
+    self._expansion_service = None
+
+  def __call__(self, *args, **kwargs):
```

Comment:
How would we preserve ordering between *args and **kwargs ?
Also we'll have to add a small change to ignore certain parameter names in Java side but this can be done (for example, arg<number>). See https://github.com/apache/beam/pull/15484#discussion_r708788543